src/map/lua/lua_baseentity.cpp

```diff
@@ -1878,6 +1996,100 @@ inline int32 CLuaBaseEntity::isFollowingPath(lua_State* L)
     return 1;
 }
 
+/************************************************************************
+* Function: rotateToAngle()
+* Purpose : rotates an entity to a given rotation
+* Example : npc:rotateToAngle(180)
+* Notes   : 
+************************************************************************/
+inline int32 CLuaBaseEntity::rotateToAngle(lua_State* L)
+{
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    TPZ_DEBUG_BREAK_IF(lua_isnil(L, 1) || !lua_isnumber(L, 1));
+
+    uint8 wantedRotation = (uint8)lua_tointeger(L, 1);
+
+    m_PBaseEntity->PAI->PathFind->RotateTo(wantedRotation);
+
+    return 0;
+}
+
+/************************************************************************
+* Function: pathStop()
+* Purpose : stops an npc from pathing until pathResume() is used
+* Example : npc:pathStop()
+* Notes   : used in conjunction with pathResume()
+************************************************************************/
+
+inline int32 CLuaBaseEntity::pathStop(lua_State* L)
+{
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity->objtype == TYPE_PC);
+
+    CBattleEntity* PBattle = (CBattleEntity*)m_PBaseEntity;
```

Comment:
`m_PBaseEntity->speed =0;` actually does work, no need to recast as a battleentity for either mobs or npc's (or players). Which makes this entire function in its current form a duplicate of `speed()`

Lets improve it by having this function also store the entities current speed to a localVar prior to setting it zero. Then resume path can read that value instead of querying the database every time.

```c++
 m_PBaseEntity->SetLocalVar(lua_tostring(L, 1), (uint64_t)lua_tointeger(L, 2));
```

Reply:
do we want to place all these in an issue or ?

Reply:
not particularly no. I'd rather see it be changed here and its such a small correction to make instead of unnecessarily adding more iops of our inefficient db connection.